### ISSUE TITLE
Write some documentation for async/futures

### ISSUE THREAD

**Issue by badboy** [mozilla/uniffi-rs#1608]
Looks like we only have examples, but nothing on https://mozilla.github.io/uniffi-rs/

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/UNIFFI-287)
┆Issue Number: UNIFFI-287

**Comment by mhammond** [comment]
> Looks like we only have examples

I don't think we even have examples - we have fixtures, but they aren't particularly approachable for understanding how async works here.

**Comment by badboy** [comment]
> > Looks like we only have examples
> 
> I don't think we even have examples - we have fixtures, but they aren't particularly approachable for understanding how async works here.

`s/examples/tests` -- is what I meant. And yeah, they are definitely not good enough for users.
